detection-rules/attachment_microsoft_image_lure_qr_code.yml

```diff
@@ -0,0 +1,43 @@
+name: "Attachment: Microsoft image with QR code"
+description: |
+  Detects messages using Microsoft image based lures, referencing a QR code from an Unsolicited sender.
+  These messages often lead users to phishing sites or initiate unwanted downloads upon scanning.
+type: "rule"
+severity: "medium"
+source: |
+  type.inbound
+  and any(attachments, .file_type  in~ ('png', 'jpeg', 'jpg', 'bmp'))
+  and any(attachments, 
+          .file_type in~ ('bmp', 'png', 'jpg', 'jpeg')
+          and any(beta.logo_detect(.).brands, .name in~ ("Microsoft", "Office365"))
+  )
+  and any(attachments,
+      .file_type in~ ('bmp', 'png', 'jpg', 'jpeg')
+      and (
+          any(file.explode(.),
+              any(.scan.strings.strings, regex.icontains(.,  'scan|camera'))   
+              and any(.scan.strings.strings, regex.icontains(.,  '\bQR\b|Q\.R\.|barcode'))
+          )
+      )
+  )
+  and (
+      not any(headers.hops, .authentication_results.compauth.verdict is not null
+      and .authentication_results.compauth.verdict == "pass"
+      and sender.email.domain.domain == "microsoft.com")
```

Comment:
```suggestion
      not any(headers.hops,
          .authentication_results.compauth.verdict is not null
          and .authentication_results.compauth.verdict == "pass"
          and sender.email.domain.domain == "microsoft.com"
      )
```

Reply:
if only we had... some way... to do this automagically